### PR TITLE
Add `Checkbox` functional tests.

### LINKED ISSUE #150: CheckBox: Add Functional Tests

### PR DESCRIPTION
**Pull request by lzhoucs**
**Type:** functional tests

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

Resolves #150

### CODE REVIEW

File: src/checkbox/tests/functional/Checkbox.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+function nthCheckbox(n: number) {
+	return `fieldset label:nth-of-type(${n}) input[type="checkbox"]`;
+}
+registerSuite({
+	name: 'Checkbox',
+
+	'checkbox should be visible'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector(nthCheckbox(1))
+		// as per mwistrand, `isDisplayed` is broken in Safari 10, commented out for now
```
Comment (dylans) [inline src/checkbox/tests/functional/Checkbox.ts]: Same note here, let's either find an external issue or create one in meta for fixing in the future when Safari driver improves.

File: src/checkbox/tests/functional/Checkbox.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	'disabled checkbox should not be toggle-able'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector(nthCheckbox(2))
+			.isSelected()
+			.then((checked: boolean) => {
+				assert.isFalse(checked, 'Initial state should be false');
+			})
+			.click()
+			.then(undefined, (err: Error) => {
+				// this then error callback is needed in FireFox with FirefoxDriver since it throws an
```
Comment (dylans) [inline src/checkbox/tests/functional/Checkbox.ts]: "this then error callback"?  Also link to a known issue please.

File: src/checkbox/tests/functional/Checkbox.ts (hunk 1 above)
Comment (lzhoucs) [inline src/checkbox/tests/functional/Checkbox.ts]: Issue link is added.

File: src/checkbox/tests/functional/Checkbox.ts (hunk 2 above)
Comment (lzhoucs) [inline src/checkbox/tests/functional/Checkbox.ts]: Issue link is added. Thanks!

**Comment by lzhoucs** [comment]
@tomdye @dylans , thanks for the review. I've addressed your feedback. Please take another look when you get chance and let me know if there's any other issues. Thanks!

**Pull request merged**